### PR TITLE
Bumped to Swift 4.0 version

### PR DESCRIPTION
**Pull request by jane-austin**
There were some problems with an [expression's complexity](https://github.com/Velhotes/Vinyl/compare/master...jane-austin:oss/swift4?expand=1#diff-c99258d9a795eb54dffe4ffa8b48a834R81), so I tried to do "dumb" it down a bit. 😿

### CODE REVIEW

**Comment by jane-austin** [comment]
This closes #77

File: Vinyl/Extensions/Bundle.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import Foundation
 
 // Idea taken from Venmo/DVR (thanks!)
-func testingBundle() -> Bundle {
+public func testingBundle() -> Bundle {
```
Comment (mluisbrown) [inline Vinyl/Extensions/Bundle.swift]: Why was this made `public` ?

File: VinylTests/Arbitrary.swift [hunk 2]
```diff
@@ -11,9 +11,9 @@ import SwiftCheck
 
 /// Generates an array of lowercase alphabetic `Character`s.
 let lowerStringGen =
-Gen<Character>.fromElements(in: "a"..."z")
-    .proliferateNonEmpty
-    .map(String.init)
+    Gen<Character>.fromElements(in: "a"..."z")
+        .proliferateNonEmpty
+        .map(String.init)
```
Comment (mluisbrown) [inline VinylTests/Arbitrary.swift]: I get a compiler error here `ambiguous use of 'init'` which I can only resolve by changing to `.map() { String.init($0) }`.  Do you not get this error?

Error detail:
```
/Users/michael/Documents/Development/github/Vinyl/VinylTests/Arbitrary.swift:16:14: error: ambiguous use of 'init'
        .map(String.init)
             ^
Swift.String:32:12: note: found this candidate
    public init<T>(stringInterpolationSegment expr: T) where T : CustomStringConvertible
           ^
Swift.String:16:12: note: found this candidate
    public init<S>(_ characters: S) where S : Sequence, S.Element == Character
           ^
```

File: VinylTests/Arbitrary.swift [hunk 3]
```diff
@@ -78,7 +78,13 @@ let pathParameterGenArray: Gen<[String]> = Gen.sized { sz in
 }
 /// Generates a set of parameters.
 let pathParameterGen: Gen<String> = pathParameterGenArray.map { xs in
-        return xs.reduce("?") { $0 == "?" ? "?" + $1 : $0 + "&" + $1
+    return xs.reduce("?") { (acc, val) in
```
Comment (mluisbrown) [inline VinylTests/Arbitrary.swift]: 😢

**Comment by mluisbrown** [comment]
This PR has been superseded by #81.

**Pull request closed without merging**